pkg/agent/nodeportlocal/k8s/controller.go

```diff
@@ -112,6 +122,107 @@ func (c *Controller) EnqueueObj(obj interface{}) {
 	c.queue.Add(podKey)
 }
 
+func (c *Controller) checkDeletedSvc(obj interface{}) (*corev1.Service, error) {
+	deletedState, ok := obj.(cache.DeletedFinalStateUnknown)
+	if !ok {
+		return nil, fmt.Errorf("Received unexpected object: %v", obj)
+	}
+	svc, ok := deletedState.Obj.(*corev1.Service)
+	if !ok {
+		return nil, fmt.Errorf("DeletedFinalStateUnknown object is not of type Service: %v", deletedState.Obj)
+	}
+	return svc, nil
+}
+
+func (c *Controller) EnqueueSvcUpdate(oldObj, newObj interface{}) {
+	// In case where the app selector in Service gets updated from one valid selector to another
+	// both set of pods (corresponding to old and new selector) need to be considered
+	newSvc, isSvc := newObj.(*corev1.Service)
+	if !isSvc {
+		var err error
+		newSvc, err = c.checkDeletedSvc(newObj)
+		if err != nil {
+			klog.Errorf("Got error while processing event update: %v", err)
+			return
+		}
+	}
+
+	oldSvc := oldObj.(*corev1.Service)
+	if oldSvc.ResourceVersion == newSvc.ResourceVersion {
+		return
+	}
+
+	podKeys := append(c.getPodsFromService(oldSvc), c.getPodsFromService(newSvc)...)
+	for _, podKey := range podKeys {
+		c.queue.Add(podKey)
+	}
+}
+
+func (c *Controller) EnqueueSvc(obj interface{}) {
+	svc, isSvc := obj.(*corev1.Service)
+	if !isSvc {
+		var err error
+		svc, err = c.checkDeletedSvc(obj)
+		if err != nil {
+			klog.Errorf("Got error while processing event update: %v", err)
+			return
+		}
+	}
+
+	for _, podKey := range c.getPodsFromService(svc) {
+		c.queue.Add(podKey)
+	}
+}
+
+func (c *Controller) getPodsFromService(svc *corev1.Service) []string {
+	var pods []string
+	annotations := svc.GetAnnotations()
+	if val, ok := annotations[NPLServiceAnnotation]; !ok || (ok && val != "true") {
```

Comment:
```suggestion
	if val, ok := annotations[NPLServiceAnnotation]; !ok || val != "true" {
```
because `ok` must be true if the second expression is being evaluated.

Reply:
So **labels** and label based filtering is case sensitive in k8s. Therefore, I feel it’s a good idea to make the **annotation** key/value follow similar case sensitivity as well.
In order to verify case sensitivity in labels, we can check via these commands, which give different outputs (if for instance the deployment has `foo=bar` label set)
`kubectl get deployment --selector "foo=bar"`
`kubectl get deployment --selector "foo=BAR"`
I could not find any specific examples in the antrea code, for annotation key/val case sensitivity.

Reply:
You are right, seems even annotations are typically handled as case sensitive.